tonilopezmr/day 2/ChecksumPartTwo.kt

```diff
@@ -0,0 +1,57 @@
+package day2
+
+import org.junit.Assert.*
+import org.junit.Test
+
+class ChecksumPartTwo {
+
+  fun evenlyDivisibleChecksum(of: Array<Array<Int>>): Int =
+      of.fold(emptyList<Int>()) { acc, a ->
+        acc + listOf(evenlyDivisibleDivision(a.sorted()))
+      }.reduce { acc, i -> acc + i }
+
+  fun evenlyDivisibleDivision(array: List<Int>): Int {
+    for (a in array) {
+      for (b in array.minus(a)) {
+        if (a % b == 0) {
+          return a / b
+        }
+      }
+    }
+
+    return 0
+  }
```

Comment:
Really nice and simple 💯

Reply:
Avoid Optional xD

Reply:
I was thinking about don't break the method, and use a flag to end the iteration when I find `a % b`

Reply:
I do prefer return over break over flags. But it usually depends on the situation.

Reply:
I think is a nice solution to use the return as the break flag 👍 😄